addons/mail/static/src/models/composer/composer.js

```diff
@@ -602,7 +602,15 @@ function factory(dependencies) {
         _getCommandFromText(content) {
             if (content.startsWith('/')) {
                 const firstWord = content.substring(1).split(/\s/)[0];
-                return this.env.messaging.commands.find(command => command.name === firstWord);
+                return this.env.messaging.commands.find(command => {
+                    if (command.name !== firstWord) {
+                        return false;
+                    }
+                    if (command.channel_types) {
+                        return command.channel_types.includes(this.thread.channel_type);
+                    }
```

Comment:
no `if`. `channel_types` must always be a list.

Reply:
This is not going to work with current data, when there is no list it means all channels types allowed, but when there is a list it means restrict to list elements.

Reply:
Oh ok, didn't know that. Can ignore this comment then.

Reply:
I think this is due to the fact all modules don't know the channel types added by other modules, so it's impossible to ever add all possible channels in the list we don't actually want any filter.

Reply:
But writing this message, I realized it would actually be a better approach if every module adding a channel type made sure to specify explicitly which commands are available on its channel type... But I guess that is a consideration for another time.